File: Rabbitool.Model/DTO/QQBot/Forum/Elem.cs
using Newtonsoft.Json;

namespace Rabbitool.Model.DTO.QQBot;

public class Elem
{
    [JsonProperty("text", NullValueHandling = NullValueHandling.Ignore)]
    public TextElem? Text { get; set; }

    [JsonProperty("image", NullValueHandling = NullValueHandling.Ignore)]
    public ImageElem? Image { get; set; }

    [JsonProperty("video", NullValueHandling = NullValueHandling.Ignore)]
    public VideoElem? Video { get; set; }

    [JsonProperty("url", NullValueHandling = NullValueHandling.Ignore)]
    public UrlElem? Url { get; set; }

    [JsonProperty("type")]
    public ElemTypeEnum Type { get; set; }
}

public enum ElemTypeEnum
{
    Text = 1,
    Image,
    Video,
    Url
}

File: dto/qqbot/forum/elem.go
package forum

type ElemTypeEnum int

const (
	Text ElemTypeEnum = iota + 1
	Image
	Video
	Url
)

type Elem struct {
	Text  *TextElem    `json:"text,omitempty"`
	Image *ImageElem   `json:"image,omitempty"`
	Video *VideoElem   `json:"video,omitempty"`
	Url   *UrlElem     `json:"url,omitempty"`
	Type  ElemTypeEnum `json:"type"`
}
